noir/noir-repo/tooling/nargo/src/artifacts/program.rs

```diff
@@ -67,3 +70,31 @@ impl From<ProgramArtifact> for CompiledProgram {
         }
     }
 }
+
+impl<'a> Files<'a> for ProgramArtifact {
```

Comment:
Why did you go this way rather than using the `Files` impl on the `file_map` field?

Reply:
I didn't notice that file_map was implementing it, since the `DebugArtifact` implemented it directly. Let me switch it to file_map then!

Reply:
Ahhhh i see what you mean now, implementing it for the type of file_map, will do that instead

Reply:
Hmmm that cannot be done since the type for file_map is an std one

Reply:
`DebugArtifact` implements `From<ProgramArtifact>` so we could just have the profile convert the program into a `DebugArtifact` and work with that.

Reply:
Sounds good! let me try that!

Reply:
Done!